pkg/grpc/api.go

```diff
@@ -1470,22 +1474,17 @@ func (a *api) SetActorRuntime(actor actors.Actors) {
 }
 
 func (a *api) GetMetadata(ctx context.Context, in *emptypb.Empty) (*runtimev1pb.GetMetadataResponse, error) {
-	temp := make(map[string]string)
+	temp, _ := a.extendedMetadata.MetadataGet()
 
-	// Copy synchronously so it can be serialized to JSON.
-	a.extendedMetadata.Range(func(key, value interface{}) bool {
-		temp[key.(string)] = value.(string)
-		return true
-	})
 	temp[daprRuntimeVersionKey] = a.daprRunTimeVersion
 	registeredComponents := make([]*runtimev1pb.RegisteredComponents, 0, len(a.components))
```

Comment:
`a.components' has no value, because the parameter `getComponentsFn` is not assigned to `components` when `NewAPI`

Reply:
This has been solved.